migrations/20171211_sanitize_emails.js

```diff
@@ -0,0 +1,14 @@
+/*
+  User creation saves email as lowercase, but updating an email did not.
+  Run this script to ensure all lowercased emails in db AFTER fix for updating emails is implemented.
+  This will fix inconsistent querying for an email when attempting to password reset.
+*/
+
+db.users.find().forEach(user => {
```

Comment:
This would need to filter for users who already have a value for auth.local.email (accounts that use only social media login don't have that), and would also need to update the users in batches to avoid overloading requests to the database. See some of the other migrations scripts for examples (just the more recently updated ones). However you might want to wait for feedback about the overall method first. :)

Reply:
I will wait for feedback as suggested!

Reply:
@matchajune Your suggested method is approved, so this change will be needed before the PR is merged.

Reply:
It would also be good if the update was performed only when the email address was not already fully lower case. That would speed up the migration.

Reply:
We should still update the users in batches as mentioned above, rather than trying to do them all at once - that can cause loading problems. Have a look at migrations/20180130_habit_birthday.js - I think that would be a decent example, although if other admins give you a better one, listen to them. :)  

The `query` variable can contain the search for addresses that have upper-case letters.

The `fields` parameter in the `dbUsers.find` command could limit the returned fields to just the email address.

The logic for the `set` variable can be simplified for this - all you need is setting the email address.
As a note to whichever admin runs the script: any errors output by the set command should be saved for examination - errors would happen if a new account had been created with a lower-cased version of an email address in an old account (we've already checked for examples of that but one or two more might be created before this migration is run). 

The `inc` variable and the code that uses it can be deleted.